cads_adaptors/adaptors/mars.py

```diff
@@ -45,6 +45,8 @@ def retrieve(self, request: Request) -> BinaryIO:
 
         if output.returncode:
             raise RuntimeError("The Direct MARS adaptor has crashed.")
+        if not os.path.getsize(result):
```

Comment:
Can this check go in the execute_mars method? This way it is also picked up by the MultiMarsAdaptor

Reply:
Not sure, from a quick look it might involve a bit of refactoring.
If `execute_mars` raises an error, then `self.context` does't get filled. The way the code is designed makes me think that we always want to fill it with stdout/stdout/user_visible_log

Reply:
Here is an alternative that would remove duplicated code in various adaptors: https://github.com/ecmwf-projects/cads-adaptors/pull/57